playwright-e2e/dsm/pages/participant-list-page.ts

```diff
@@ -136,20 +138,67 @@ export default class ParticipantListPage {
     return download;
   }
 
-/**
- * Filters the participant list to search for a specific participant when given their guid
- * @param participantGUID the guid of the specific participant to search for
- */
-public async filterListByParticipantGUID(participantGUID: string): Promise<void> {
-  const customizeViewPanel = this.filters.customizeViewPanel;
-  await customizeViewPanel.open();
-  await customizeViewPanel.selectColumns('Participant Columns', ['Participant ID']);
+  /**
+   * Filters the participant list to search for a specific participant when given their guid
+   * @param participantGUID the guid of the specific participant to search for
+   */
+  public async filterListByParticipantGUID(participantGUID: string): Promise<void> {
+    const customizeViewPanel = this.filters.customizeViewPanel;
+    await customizeViewPanel.open();
+    await customizeViewPanel.selectColumns('Participant Columns', ['Participant ID']);
 
-  const searchPanel = this.filters.searchPanel;
-  await searchPanel.open();
-  await searchPanel.text('Participant ID', {textValue: participantGUID });
-  await searchPanel.search();
-}
+    const searchPanel = this.filters.searchPanel;
+    await searchPanel.open();
+    await searchPanel.text('Participant ID', {textValue: participantGUID });
+    await searchPanel.search();
+  }
+
+  /**
+   * Returns the guid of the most recently created playwright participant
+   * @param isRGPStudy mark as true or false if this is being ran in RGP - parameter is only needed if method is ran in RGP study
+   * @returns the guid of the most recently registered playwright participant
+   */
+  public async getGuidOfMostRecentAutomatedParticipant(isRGPStudy?: boolean): Promise<string> {
```

Comment:
I'm confused by this func. Can you break it down into smaller pieces? Also, if it's RGP specific, why can't this added inside the `rg-pariticipant-page.ts`?

Reply:
If working with a table, it should be a object of `participant-list-table.ts`.

Reply:
will do re: smaller pieces - re: the parameter - it's not supposed to be RGP specific - the isRGPStudy is optional - for the above function I'm searching for the most recently created participant (it actually looks for participants created that day - but maybe it'll be more robust if I have it search using pts created within the past week) and from there it looks based on the first name of the participants e.g. E2E - I'm seeing that other names are used in other studies so I'll just have it check if the first name contains at least a substring of known test names - but the reason why isRGPStudy is a parameter is because it has its own separate first name column that needs to be deselected first i.e. the first name found in family member Participant Info Columns section vs general account first name found in the Participant Columns section - this is because RGP seems to have one of the most customized default filters (most other studies just use the first name in the other section since they don't have different family members attached to 1 account)

Reply:
TL;DR is RGP has an additional step to consider when searching for the most recent playwright participant when using first name to do so - so that parameter is there

Reply:
also will do re: `participant-list-table.ts`